pb_decode_fast.h

```diff
@@ -0,0 +1,1644 @@
+    do {                                                                                                                                   \
+        PB_FREE(data);                                                                                                                     \
+    } while (0)
+
+#define PB_FREE_BYTES_ATYPE_STATIC(data)
+
+#define PB_REALLOC_BYTES_ATYPE_POINTER(data, size, capacity)                                                                               \
+    do {                                                                                                                                   \
+        PB_REALLOC(data, size);                                                                                                            \
+    } while (0)
+
+#define PB_REALLOC_BYTES_ATYPE_STATIC(data, size, capacity)                                                                                \
+    do {                                                                                                                                   \
+        if (size > capacity)                                                                                                               \
+            PB_RETURN_ERROR(stream, "size too large");                                                                                     \
+    } while (0)
+
+#define PB_DECODE_VALUE_FIXED64(structname, atype, htype, ltype, fieldname, tag, data)                                                     \
+    do {                                                                                                                                   \
+        if (pb->decode_double_as_float && sizeof(*data) == 4)                                                                              \
+        {                                                                                                                                  \
+            if (!pb->decode_double_as_float(stream, (float *)data))                                                                        \
+                PB_RETURN_ERROR(stream, "failed to read wire value");                                                                      \
+        }                                                                                                                                  \
+        else if (pb->decode_fixed64)                                                                                                       \
+        {                                                                                                                                  \
+            if (!pb->decode_fixed64(stream, (double *)data))                                                                               \
+                PB_RETURN_ERROR(stream, "failed to read wire value");                                                                      \
```

Comment:
I think most of these `PB_RETURN_ERROR` could be just `return false`. The nanopb functions called should have set the error message already before returning false. In most cases, the error is coming from `pb_read()`.

`PB_RETURN_ERROR` does not overwrite existing error message, so there is no harm in this. But all these strings will use unnecessary memory and code size. It can also be confusing to follow, as you'd expect to get `"failed to read wire value"` if it fails here, but really you get the message from `pb_read()`.